### PR TITLE
Remove intermediate OpMethodCtx

### PR DESCRIPTION
**Pull request by littledivy**
It is cheaper to store all OpCtx in one place and pass around OpMethodDecl and an offset that divides the OpCtx in normal ops and method ops.

### CODE REVIEW

File: testing/unit/resource_test.ts [hunk 1]
```diff
@@ -65,8 +65,8 @@ test(async function testCppgcAsync() {
 });
 
 test(function testDomPoint() {
-  const p1 = new DOMPoint(100, 100);
   const p2 = new DOMPoint();
+  const p1 = new DOMPoint(100, 100);
```
Comment (bartlomieju) [inline testing/unit/resource_test.ts:69]: Unrelated change?

**Review by bartlomieju — APPROVED** [review]
LGTM, good catch

File: testing/unit/resource_test.ts (hunk 1 above)
Comment (littledivy) [reply testing/unit/resource_test.ts:69]: Oops, will change it back in another PR
